File: 2020/02/2.rb
i = File.readlines("input.txt").map{ s,e,l,pw = _1.scan /\w+/; [s.to_i,e.to_i,l,pw] }

p i.count{ |s,e,l,pw| (s..e).include? pw.count l }
p i.count{ |s,e,l,pw| (pw[s-1] == l) ^ (pw[e-1] == l) }

File: 2020/02/2.py
import re
pws = open("input.txt").read().split("\n")
a = b = 0
for i in pws:
    # part a
    s, e, l, p = re.split(r"\W+", i)
    x = p.count(l)
    if int(s) <= x <= int(e): a+=1
    # part b
    y = p[int(s)-1] == l
    z = p[int(e)-1] == l
    if y ^ z: b += 1
print(a, b)
